app/controllers/SessionsCategoryController.scala

```diff
@@ -0,0 +1,265 @@
+              Future.successful(BadRequest("No primary category found."))
+            } { categoryInfo =>
+              val newSubCategory = categoryInfo.subCategory.find(_.toLowerCase == cleanedSubCategoryName.toLowerCase)
+              newSubCategory.fold {
+                val subCategoryInfo = CategoryInfo(categoryName, List(cleanedSubCategoryName), categoryInfo._id)
+                categoriesRepository
+                  .upsert(subCategoryInfo)
+                  .map { result =>
+                    if (result.ok) {
+                      Logger.info(s"Sub-category was successfully added $subCategory")
+                      Ok("Sub-category was successfully added")
+                    } else {
+                      Logger.error(s"Something went wrong while adding sub-category $subCategory")
+                      BadRequest("Unsuccessful sub-category added")
+                    }
+                  }
+              } { _ =>
+                Future.successful(BadRequest("Error! Sub-category already exists"))
+              }
+            }
+        }
+    }
+  }
+
+  def modifyPrimaryCategory(categoryId: String, newCategoryName: String): Action[AnyContent] = superUserAction.async { implicit request =>
+    val cleanedCategoryName = newCategoryName.trim
+
+    if (cleanedCategoryName.isEmpty) {
+      Future.successful(BadRequest("Modify primary category cannot be empty"))
+    } else {
+      categoriesRepository
+        .getCategories
+        .flatMap { categories =>
+          val category = categories.filter(_._id.stringify == categoryId).head
+          sessionsRepository
+            .updateCategoryOnChange(category.categoryName, cleanedCategoryName)
+            .flatMap { session =>
+              if (session.ok) {
+                categoriesRepository
+                  .modifyPrimaryCategory(category._id.stringify, cleanedCategoryName)
+                  .map { result =>
+                    if (result.ok) {
+                      Logger.info(s"Primary category was successfully modified $newCategoryName")
+                      Ok("Primary category was successfully modified")
+                    } else {
+                      Logger.error(s"Something went wrong while modifying primary category $newCategoryName")
+                      BadRequest("Unsuccessful attempt to modify primary category")
+                    }
+                  }
+              } else {
+                Future.successful(BadRequest("Update on session table was unsuccessful"))
+              }
+            }
+        }
+    }
+  }
+
+  def modifySubCategory(categoryName: String,
+                        oldSubCategoryName: String,
+                        newSubCategoryName: String): Action[AnyContent] = adminAction.async { implicit request =>
+    if (newSubCategoryName.trim.isEmpty) {
+      Future.successful(BadRequest("Modify sub-category cannot be empty"))
```

Comment:
Replace "Modify sub-category cannot be empty" with "Modified sub-category cannot be empty"